eth_account/_utils/typed_transactions.py

```diff
@@ -345,3 +334,158 @@ def vrs(self) -> Tuple[int, int, int]:
+            # This is insane logic that is required because the way we evaluate
+            # correct types is in the `if not all()` branch below, and 0 obviously
+            # maps to the int(0), which maps to False... This was not an issue in non-typed
+            # transaction because v=0, couldn't exist with the chain offset.
+            dictionary['v'] = '0x0'
+        valid_fields = apply_formatters_to_dict(
+            transaction_valid_values, dictionary,
+        )  # type: Dict[str, Any]
+        if not all(valid_fields.values()):
+            invalid = {key: dictionary[key] for key, valid in valid_fields.items() if not valid}
+            raise TypeError("Transaction had invalid fields: %r" % invalid)
+
+    @classmethod
+    def from_dict(cls, dictionary: Dict[str, Any]):
+        """
+        Builds a DynamicFeeTransaction from a dictionary.
+        Verifies that the dictionary is well formed.
+        """
+        # Validate fields.
+        cls.assert_valid_fields(dictionary)
+        sanitized_dictionary = pipe(
+            dictionary,
+            dict,
+            partial(merge, cls.transaction_field_defaults),
+            apply_formatters_to_dict(TYPED_TRANSACTION_FORMATTERS),
+        )
+
+        # We have verified the type, we can safely remove it from the dictionary,
+        # given that it is not to be included within the RLP payload.
+        transaction_type = sanitized_dictionary.pop('type')
+        if transaction_type != cls.transaction_type:
+            raise ValueError(
+                "expected transaction type %s, got %s" % (cls.transaction_type, transaction_type),
+            )
+        return cls(
+            dictionary=sanitized_dictionary,
+        )
+
+    @classmethod
+    def from_bytes(cls, encoded_transaction: HexBytes):
+        """Builds a DynamicFeeTransaction from a signed encoded transaction."""
+        if not isinstance(encoded_transaction, HexBytes):
+            raise TypeError("expected Hexbytes, got type: %s" % type(encoded_transaction))
+        if not (len(encoded_transaction) > 0 and encoded_transaction[0] == cls.transaction_type):
+            raise ValueError("unexpected input")
+        # Format is (0x02 || TransactionPayload)
+        # We strip the prefix, and RLP unmarshal the payload into our signed transaction serializer.
+        transaction_payload = encoded_transaction[1:]
+        rlp_serializer = cls._signed_transaction_serializer
+        dictionary = rlp_serializer.from_bytes(transaction_payload).as_dict()  # type: ignore
+        dictionary['type'] = cls.transaction_type
+        return cls.from_dict(dictionary)
+
+    def as_dict(self) -> Dict[str, Any]:
+        """Returns this transaction as a dictionary."""
+        dictionary = self.dictionary.copy()
+        dictionary['type'] = self.__class__.transaction_type
+        return dictionary
+
+    def hash(self) -> bytes:
+        """
+        Hashes this AccessListTransaction to prepare it for signing.
```

Comment:
copy/paste edit: `AccessListTransaction` -> `DynamicFeeTransaction`

Also we can edit the message here to not include `gasPrice` or `accessList` in the `rlp()` here or refactor this message in another way to avoid confusion